### ISSUE TITLE
Opening and reading a file repeatadly is unreliable:

### ISSUE THREAD

**Issue by vi** [whoozle/android-file-transfer-linux#113]
```
$ fuse/aft-mtp-mount m -d
...
unique: 168, opcode: READ (15), nodeid: 1047181, insize: 80, pid: 4624
   Read 1047181 131072 917504
   unique: 168, success, outsize: 131088
unique: 169, opcode: READ (15), nodeid: 1047181, insize: 80, pid: 4624
   Read 1047181 131072 1048576
   unique: 169, success, outsize: 131088
unique: 170, opcode: READ (15), nodeid: 1047181, insize: 80, pid: 4624
   Read 1047181 131072 1179648
   unique: 170, success, outsize: 131088
unique: 171, opcode: RELEASE (18), nodeid: 1047181, insize: 64, pid: 0
   Release 1047181
unique: 172, opcode: GETATTR (3), nodeid: 1047181, insize: 56, pid: 4626
   GetAttr 1047181
   unique: 172, success, outsize: 120
unique: 173, opcode: OPEN (14), nodeid: 1047181, insize: 48, pid: 4626
   Open 1047181
   unique: 173, success, outsize: 32
{frozen}
^C
```

```
$ for((;;)) { pv -L 400k m/Internal\ storage/Music/q.tta | dd bs=1M count=1 iflag=fullblock &> /dev/null; }
1.02MB 0:00:02 [ 391kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 385kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 387kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 399kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 388kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 399kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 400kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 387kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 400kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 388kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 390kB/s] [>                                                                                                                           ]  0%            
1.02MB 0:00:02 [ 397kB/s] [>                                                                                                                           ]  0%            
pv: (stdin): read failed: Transport endpoint is not connected
   0B 0:00:49 [   0B/s] [>                                                                                                                             ]  0%           
```

**Comment by vi** [comment]
Tried the second time: also 12 times successful file open and read and 13'th time is hang.

4e16a115a9941f1a9f99a5a135414f8eb59da7c2

**Comment by whoozle** [comment]
i was able to reproduce it with the following command:
```find /path/to/device -type f -exec dd if={} of=/dev/null \;```

**Comment by whoozle** [comment]
the problem is in async_read. investigating.

**Comment by whoozle** [comment]
Fixed in 3d3843bd9e97729328796eb35e895b044a25e809

**Comment by vi** [comment]
> my average speed is more promising

Note `-L 400k` in my command line. That's intentional speed limiter.
